From d58a0ac4a379e3563b8983bea55f4cdfd0a833d1 Mon Sep 17 00:00:00 2001
From: vagin ivan <vagin.ivan1@huawei-partners.com>
Date: Wed, 13 Jul 2022 12:33:38 +0300
Subject: [PATCH] Migrate from clang-9 to clang-14

---
 BUILD.gn       | 4 ++++
 CMakeLists.txt | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/BUILD.gn b/BUILD.gn
index 6ca8809..20553ef 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -32,6 +32,10 @@ config("vixl_public_config") {
     defines = [ "VIXL_DEBUG" ]
   }
 
+  if (ark_standalone_build) {
+    cflags_cc = [ "-Wno-bitwise-instead-of-logical" ]
+  }
+
   include_dirs = [
     "$ark_third_party_root/vixl",
     "$ark_third_party_root/vixl/src",
diff --git a/CMakeLists.txt b/CMakeLists.txt
index f2ac6c6..249992e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -101,6 +101,10 @@ target_link_libraries(vixl arkbase)
 target_compile_options(vixl PUBLIC ${VIXL_FLAGS})
 target_compile_options(vixl PRIVATE -Wno-shadow)
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 14.0.0)
+    target_compile_options(vixl PRIVATE -Wno-bitwise-instead-of-logical)
+endif()
+
 set_property(TARGET vixl PROPERTY POSITION_INDEPENDENT_CODE ON)
 set_property(TARGET vixl PROPERTY EXCLUDE_FROM_ALL TRUE)
 
-- 
2.25.1

